### PR TITLE
[VBLOCKS-1417] Outgoing call screen and active call screen

### PR DESCRIPTION
**Pull request by mhuynh5757**
This pull request includes the outgoing call screen and active call screen part 1 (without active call banner).

https://issues.corp.twilio.com/browse/VBLOCKS-1417

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Review by charliesantos — CHANGES_REQUESTED** [review]
Initial review on the app/ level files.

File: app/__tests__/store.test.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  });
+  await app.store.dispatch(makeOutgoingCallAction);
+
+  console.log(app.store.getState());
+
+  const muteActiveCallAction = activeCall.muteActiveCall({ mute: true });
+  await app.store.dispatch(muteActiveCallAction);
+
+  console.log(app.store.getState().voice.call.outgoingCall);
+});
```
[the remark this answers is not in the document]
Comment (mhuynh5757) [reply app/__tests__/store.test.ts:82]: Yeah, I wanted to test at least the happy path for the application state. I think we're going to want to use something like Storybook down the line.

**Comment by charliesantos** [comment]
I reviewed the app code and kudos to you @mhuynh5757 . This is a lot of work! However, there's an issue I noticed and maybe you can help me understand. We are using redux, but I cannot see/understand the pattern that you're trying to do. The way it's currently setup, the data flow looks like this
```
Screens/Components <------> State/Store
```

On a redux pattern, I'm expecting actions, reducers, and components that interacts with the state/store in the following manner
```
Components ----> Dispatch Actions ----> Store Reducers ----> State ---> Components
```
And have folder structure that maps to this data flow - `components`, `actions`, `reducers`. I noticed you have `screens` folder, which is fine btw.

My request is, please explain what pattern you're trying to use. If it's not industry standard or something you came up with, please explain why. We need to understand pros and cons.

**Comment by charliesantos** [comment]
Notes and action items base on our discussion:
- Move the hooks into separate files/folders
- Actions and Reducers are good. Need to document the pattern and file structure we're using
- Rename `Other` component to something more meaningful
- Document the distinction between screens and components
We should submit a ticket to address the action items. But we need to rename `Other` component now :)

File: app/README.md [hunk 2]
```diff
@@ -1,66 +1,40 @@
...
 
-- [React Native CLI](https://reactnative.dev/docs/environment-setup)
-- [Yarn](https://yarnpkg.com/getting-started)
-- Token - Create a file, `e2e-tests-token.ts` under `example/src` directory with the following content.
-  ```ts
-  export const token = 'my-token-goes-here';
-  ```
+The Reference App showcases the following possibilities using Twilio Voice:
+* Dialpad for making outgoing calls
+* TODO
```
[the remark this answers is not in the document]
Comment (mhuynh5757) [reply app/README.md:9]: Looked at the PRD again, didn't see `Contact List` so I'll leave it out of the README for now. Let me know if we should add it anyways.

File: app/README.md [hunk 3]
```diff
@@ -1,66 +1,40 @@
-# Twilio Voice React Native App
+# Reference App
```
[the remark this answers is not in the document]
Comment (mhuynh5757) [reply app/README.md]: I won't resolve this one yet since we have work slated for branding down the line, leaving it for posterity.

**Review by charliesantos — APPROVED** [review]

**Pull request merged**